languages/csharp/exercises/concept/lists/.meta/design.md

```diff
@@ -0,0 +1,30 @@
+## Learning objectives
+
+- Know what generic types are.
+- Know of the existence of the `List<T>` type.
+- Know how a list is different from an array.
+- Know how to define a list.
+- Know how to add and remove elements from a list.
+- Know how to access elements in a list by their index.
+- Know how to iterate over elements in a list.
+- Know some basic list functions (like finding the index of an element in a list or sorting a list).
+
+## Out of scope
+
+- Generic functions.
+- Generic constraints.
+- Memory and performance characteristics.
+- LINQ.
+- Concurrency issues.
+- Co-/contravariance.
+- Equality.
+- List resizing due to it using an array as its data type.
+
+## Concepts
+
+- `lists`: know of the existence of the `List<T>` type; know how a list is different from an array; know how to define a list; know how to add and remove elements from a list; know how to access elements in a list by their index; know how to iterate over elements in a list; know some basic list functions (like finding the index of an element in a list).
+- `generic-types`: know what generic types are.
+
+## Prerequisites
+
+- `for-loops`: know how to use a `for-loop` to iterate over a collection.
```

Comment:
I thought that the `arrays` concept was missing too, as the above `lists` concept states "know how a list is different from an array". The `for-loops` concept is introduced with arrays, so technically this is not necessary.

Reply:
added to _design.md_